wordpress-mobile/WordPress-Android
# Add a guard to avoid calling Glide load when activity is destroyed

Fixes #18809

Related PR:

https://github.com/Automattic/stories-android/pull/739

### Description

This PR adds a guard to avoid calling Glide's load when the activity is destroyed, which otherwise leads to a crash on a failed assertion. [More details in the issue](https://github.com/wordpress-mobile/WordPress-Android/issues/18809#issuecomment-1651126958).

Note: I was unable to reproduce the underlying issue, so I was unable to confirm that this changeset will fix the issue. Nevertheless, if the issue persists after landing this, it may help narrow the search for the root cause.

<details>
<summary>Attempts to reproduce</summary>

In order to attempt reproducing this issue, I tried several combinations of:

* Turning on "Don't Keep Activities" in developer settings
* Setting breakpoints at various points in the flow to try simulating a race condition

Though I was unable to successfully trigger a race condition, it is still possible it is occurring in production, since some race conditions might be hard to simulate without deeply debugging sources for framework and / or library code, and debugging different threads separately.

</details>

To test:

1. Log into an account and select a site with story posts enabled
2. Add a new stories post (and select an image)
3. Expect that the image is displayed correctly (in the foreground and background)

Bonus:

1. Add more picture slides
2. Switch between the story slides
3. Make sure that the images and background are displayed correctly for each image slide


## Regression Notes
1. Potential unintended areas of impact
Stories image loading

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
Setting up the necessary foundation to create a UI test for this flow is out of scope for this task.

PR submission checklist:


UI Changes testing checklist:
